src/main/java/step2/StringCalculator.java

```diff
@@ -0,0 +1,106 @@
+package step2;
+
+import java.util.regex.Pattern;
+
+public class StringCalculator {
+
+    private static final String SEPARATOR = " ";
+
+    public static void main(String[] args) {
+        StringCalculator stringCalculator = new StringCalculator();
+        int result = stringCalculator.calculate("2 + 3 * 4 / 2");
+        System.out.println(result);
+    }
+
+    public int calculate(String input) {
+
+        //check
+        inputCheck(input);
+
+        // ready
+        String[] inputArray = input.split(SEPARATOR);
+        int result = Integer.parseInt(inputArray[0]);
+
+        // calculate
+        for (int i = 0; i < inputArray.length - 1; i += 2) {
+            result = plus(result, Integer.parseInt(inputArray[i + 2]), inputArray[i + 1]);
+            result = minus(result, Integer.parseInt(inputArray[i + 2]), inputArray[i + 1]);
+            result = multiply(result, Integer.parseInt(inputArray[i + 2]), inputArray[i + 1]);
+            result = division(result, Integer.parseInt(inputArray[i + 2]), inputArray[i + 1]);
```

Comment:
하나의 연산을 하기 위해 사칙연산 체크를 모두 해봐야하는군요..! 만약 사칙연산 외에 연산이 추가되면, 연산량이 더 많아지게 되는 구조 같습니다 ㅜ 전체적인 구조 변경이 필요해보입니다!

Reply:
네 메소드를 각자 자신의 역할에 맞게 동작하게끔 최소한으로 줄여서 해보겠다는 욕심에 매번 호출해서 연산자 비교 후 실동작을 하면 되겠다 싶었는데, 하나의 메소드가 해당 동작을 할 수 있게 바꾸어 보겠습니다.
```suggestion
result = calculateProcess(result, Integer.parseInt(inputArray[i + 2]), inputArray[i + 1]);
```